full-stack-engineer/submissions/michaelyao/client/src/App.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+import { CssBaseline } from '@mui/material';
+import AppRoutes from './AppRoutes';
+import { AuthContext } from './contexts/authContext';
+import { currentUser } from './Api';
+
+function App() {
+  const [user, setUser] = useState(null);
+  const value = {
+    user,
+    setUser,
+  };
+
+  useEffect(() => {
+    // When the app is lauched, let's check the user is signed in or not.
+    const fetchData = async () => {
+      try {
+        const { data } = await currentUser();
+        setUser(data);
+      } catch (error) {}
```

Comment:
Is this catch empty because setUser `data` is supposed to fail if not logged (I will get this code soon) and silent in purpose?
I would suggest don't generating an error if a user is not logged, but notify if some error happens in API call.

Reply:
Yep. Here, the app is trying to sign in the user from cookies once the app is mounted. If it can not get the current user, then it will keep silent. But I think it's better to setUser to `null` to make it more readable. I updated it already.